jina/drivers/index.py

```diff
@@ -38,7 +38,18 @@ class KVIndexDriver(BaseIndexDriver):
     """Serialize the documents/chunks in the request to key-value JSON pairs and write it using the executor
     """
 
+    def __init__(self, id_tag: Optional[str] = None, *args, **kwargs):
+        """ Initialize a :class:`BaseExecutableDriver`
+
+        :param id_tag: Optional parameter if wants to index documents based on a specific tag instead of the `document` id field
+        """
+        super().__init__(*args, **kwargs)
+        self.id_tag = id_tag
+
     def _apply_all(self, docs: Iterable['jina_pb2.Document'], *args, **kwargs) -> None:
-        keys = [uid.id2hash(doc.id) for doc in docs]
+        if self.id_tag:
+            keys = [doc.tags[self.id_tag] for doc in docs]
+        else:
+            keys = [uid.id2hash(doc.id) for doc in docs]
```

Comment:
hmm... changing the existing driver when adding new driver. something o avoid

Reply:
I agree. Might it be a good idea to add another `TagKVIndexDriver` instead?

Reply:
I think it can be useful, but I will revert it to jsut using `id` and we can handle that in another PR

Reply:
Reverted to only having `doc.id` and its hash as key, we can think about adding another driver in the future or another PR